core/src/test/scala/org/locationtech/rasterframes/encoders/DevSpec.scala

```diff
@@ -0,0 +1,270 @@
+      }
+
+      val teacher = Person("Sid", 0)
+      val student = Person("Eugene", 10)
+      val school = School(teacher, student)
+      val ds = Seq(school).toDS
+
+      ds.printSchema()
+      ds.show()
+
+      val df = ds.toDF()
+
+      val it = df.as[School].first()
+      info(it.toString)
+    }
+  }
+
+  describe("scalar tile operations") {
+    import org.locationtech.rasterframes.tiles.ProjectedRasterTile
+
+    import spark.implicits._
+
+    ignore("=== DeSer for Time ===") {
+      import org.apache.spark.sql.catalyst.ScalaReflection.deserializerForType
+      import scala.reflect.runtime.universe._
+
+      val de = deserializerForType(typeOf[java.sql.Timestamp])
+
+      info(de.numberedTreeString)
+    }
+
+
+    ignore("=== DeSer for Person ===") {
+      import org.apache.spark.sql.catalyst.ScalaReflection.deserializerForType
+      import scala.reflect.runtime.universe._
+
+      val de = deserializerForType(typeOf[Person])
+
+      info(de.numberedTreeString)
+    }
+
+    it("Person Frame"){
+      val ds = Seq((Person("Bob", 11), Person("Eugene", 10))).toDS
+      ds.printSchema()
+      ds.show()
+    }
+
+    ignore("does it really serialize CRS?") {
+      val first = CRSEncoder()
+
+      info("Before: \n" + first.deserializer.treeString)
+
+      val en = first.resolveAndBind()
+      info("After: \n" + en.deserializer.treeString)
+
+      val ir = en.createSerializer().apply(LatLng)
+      val out = en.createDeserializer().apply(ir)
+      out shouldBe LatLng
+    }
+
+    it("person") {
+      val data = List(Person("Eugene", 39), Person("Bob", 45))
+      val ds = data.toDS
+      ds.printSchema()
+      ds.show()
+      val df = ds.toDF()
+
+      val me = df.as[Person].first()
+      info(me.toString)
+    }
+
+    it("person encoder"){
+      val en = ExpressionEncoder[Person]
+      info(en.objSerializer.treeString)
+      info(en.deserializer.treeString)
+    }
+
+    it("prt encoder"){
+      val en = ProjectedRasterTile.prtEncoder
+      info(en.objSerializer.treeString)
+      info(en.deserializer.treeString)
+    }
+
+    it("should round trip ProjectedRasterTile") {
+      val data = Seq(one, two)
+      val ds = data.toDS
+      ds.printSchema()
+      ds.show()
+      val df = ds.toDF()
+
+      val tile = df.as[ProjectedRasterTile].first()
+      info(tile.toString)
+    }
+
+    it("one") {
+      type T = ProjectedRasterTile
+      val data = Seq(one)
+      val in: Encoder[T] = implicitly[Encoder[T]]
+      val enc = encoderFor[T](in)
+      val toRow = enc.createSerializer()
+      val attributes = enc.schema.map(f => AttributeReference(f.name, f.dataType, f.nullable, f.metadata)())
+      val encoded = data.map(d => toRow(d).copy())
+      val plan = new LocalRelation(attributes, encoded)
+      val dataset = new Dataset(spark, plan, implicitly[Encoder[T]])
+      dataset.printSchema()
+      dataset.show()
```

Comment:
This breaks out the series of events that result 
In Spark 3 the output is:
```
root
 |-- tile_context: struct (nullable = true)
 |    |-- extent: struct (nullable = false)
 |    |    |-- xmin: double (nullable = false)
 |    |    |-- ymin: double (nullable = false)
 |    |    |-- xmax: double (nullable = false)
 |    |    |-- ymax: double (nullable = false)
 |    |-- crs: struct (nullable = false)
 |    |    |-- crsProj4: string (nullable = true)
 |-- tile: tile (nullable = false)

+--------------------+--------------------+
|        tile_context|                tile|
+--------------------+--------------------+
|{{10.0, 20.0, 30....|ArrayTile(10,10,i...|
```

In spark 2:

```
root
 |-- value: struct (nullable = true)
 |    |-- tile_context: struct (nullable = true)
 |    |    |-- extent: struct (nullable = false)
 |    |    |    |-- xmin: double (nullable = false)
 |    |    |    |-- ymin: double (nullable = false)
 |    |    |    |-- xmax: double (nullable = false)
 |    |    |    |-- ymax: double (nullable = false)
 |    |    |-- crs: struct (nullable = false)
 |    |    |    |-- crsProj4: string (nullable = false)
 |    |-- tile: tile (nullable = false)

+--------------------+
|               value|
+--------------------+
|[[[10.0, 20.0, 30...|
|[[[10.0, 20.0, 30...|
+--------------------+
```

Reply:
This behavior breaks `.as[ProjectedRasterTile]` both when used on the output of DataFrame and when used on the column type conversion from DataFrame to DataSet. It seems to be the root cause of many/most/all failed tests where assignment to `Tuple1` during expression evaluation fails because ProjectedRasterTile gets blown out in two columns instead of one.

Reply:
Little more context here: It seems that "value" nesting is now reserved for things that aren't `StructType` (primitives and arrays) https://github.com/apache/spark/blob/1d550c4e90275ab418b9161925049239227f3dc9/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala#L262-L267

Quick experiment of wrapping the ProjectedRasterTile in `value` struct on our end yielded predictable results of fixing the "one" test and breaking the "two" test but preserving the intermediate `value` nesting under each column.